### PR TITLE
Namespace list

### PR DESCRIPTION
**Pull request by kwiatekus**
**Description**

Changes proposed in this pull request:
- Add namespaces list
- As react logs are served from different address, app does not have access to original core's localStorage (like `showSystemNamespaces`). This should be resolved when ng-core are fully replaced with react core.
- TODO: update backend (add healthy pods counting, namespace name etc.) in other task and change todos in code.

**Related issue(s)**
https://github.com/kyma-project/console/issues/1089

### CODE REVIEW

File: core-ui/src/components/NamespaceList/NamespaceList.js [hunk 1]
```diff
@@ -1,13 +1,84 @@
...
+          result = [...currentItems, targetItem];
+        }
+        break;
+      case 'UPDATE':
+        idx = items.findIndex(i => i.name === targetItem.name);
+        if (idx === -1) {
+          result = [...currentItems, targetItem];
+          break;
+        }
+        items[idx] = targetItem;
```
Comment (sjanota) [inline core-ui/src/components/NamespaceList/NamespaceList.js]: Are you sure mutable update triggers rerender? IT probably should be something like:
```javascript
items = [...items.slice(0, idx-1), targetItem, ...items.slice(idx+1, items.length)]
```

File: core-ui/src/components/NamespaceList/NamespaceList.js [hunk 2]
```diff
@@ -1,13 +1,84 @@
...
+    localStorage.getItem('console.showSystemNamespaces'),
+  );
+  const { data, error, loading, subscribeToMore } = useQuery(GET_NAMESPACES, {
+    variables: {
+      showSystemNamespaces,
+    },
+  });
+
+  const handleNamespaceEvent = (obj = {}, event = {}) => {
+    const currentItems = obj.namespaces || [];
```
Comment (sjanota) [inline core-ui/src/components/NamespaceList/NamespaceList.js]: Can it even happen, that we don't have namespaces yet?

File: core-ui/src/queries/queries.js [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
       applications
+      isSystemNamespace
+    }
+  }
+`;
+
+export const NAMESPACES_EVENT_SUBSCRIPTION = gql`
+  subscription Namespaces($showSystemNamespaces: Boolean) {
+    namespaceEvent(withSystemNamespaces: $showSystemNamespaces) {
+      type
```
Comment (sjanota) [inline core-ui/src/queries/queries.js]: Extract the query content as a fragment. https://www.apollographql.com/docs/react/data/fragments/

File: core-ui/src/components/NamespaceList/NamespacesGrid/NamespaceDetailsCard/NamespaceDetailsCard.scss [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    .overlay {
+        pointer-events: none;
+        background-color: rgba(225, 225, 225, 0.3);
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        left: 0;
+        top: 0;
+        z-index: 100000;
```
Comment (parostatkiem) [inline core-ui/src/components/NamespaceList/NamespacesGrid/NamespaceDetailsCard/NamespaceDetailsCard.scss]: more

File: core-ui/src/components/NamespaceList/NamespacesGrid/NamespaceDetailsCard/test/NamespaceDetailsCard.test.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+          applicationsCount={0}
+        />
+      </MockedProvider>,
+    );
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+
+    expect(console.warn.mock.calls.length).toBe(2);
+    expect(console.warn.mock.calls[0][0]).toMatchSnapshot();
+    expect(console.warn.mock.calls[0][1]).toMatchSnapshot();
```
Comment (parostatkiem) [inline core-ui/src/components/NamespaceList/NamespacesGrid/NamespaceDetailsCard/test/NamespaceDetailsCard.test.js]: why not 
```javascript
    expect(console.warn.mock.calls).toMatchSnapshot();
```
?

File: core-ui/src/components/NamespaceList/NamespacesListHeader/NamespaceFilters/NamespaceFilters.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  const popoverBody = (
+    <Menu.List>
+      {Array.from(filters).map(e => (
+        <Menu.Item key={e.name} onClick={() => onFilterClick(e)}>
+          <FormInput
+            type="checkbox"
+            className="fd-has-margin-right-small"
+            checked={e.isSelected}
+            onChange={() => {
+              /* event will be propagated to Menu.Item */
```
Comment (parostatkiem) [inline core-ui/src/components/NamespaceList/NamespacesListHeader/NamespaceFilters/NamespaceFilters.js]: So does it need to be here?

**Pull request merged**